proposals/0029-cooperative-vector.md

```diff
@@ -602,92 +615,90 @@ The vector-matrix intrinsics are expected to be supported in all shader stages.
-  D3D12_COOPERATIVE_VECTOR_DATATYPE_UINT16          =  3, // ComponentType::U16
-  D3D12_COOPERATIVE_VECTOR_DATATYPE_SINT32          =  4, // ComponentType::I32
-  D3D12_COOPERATIVE_VECTOR_DATATYPE_UINT32          =  5, // ComponentType::U32
-  D3D12_COOPERATIVE_VECTOR_DATATYPE_FLOAT16         =  7, // ComponentType::F16
-  D3D12_COOPERATIVE_VECTOR_DATATYPE_FLOAT32         =  8, // ComponentType::F32
-  D3D12_COOPERATIVE_VECTOR_DATATYPE_INT8_T4_PACKED  = 16, // ComponentType::PackedS8x32
-  D3D12_COOPERATIVE_VECTOR_DATATYPE_UINT8_T4_PACKED = 17, // ComponentType::PackedU8x32
-  D3D12_COOPERATIVE_VECTOR_DATATYPE_UINT8           = 18, // ComponentType::U8
-  D3D12_COOPERATIVE_VECTOR_DATATYPE_SINT8           = 19, // ComponentType::I8
-  D3D12_COOPERATIVE_VECTOR_DATATYPE_E4M3            = 20, // ComponentType::F8_E4M3 (1 sign, 4 exp, 3 mantissa bits)
-  D3D12_COOPERATIVE_VECTOR_DATATYPE_E5M2            = 21, // ComponentType::F8_E5M2 (1 sign, 5 exp, 2 mantissa bits)
+typedef enum D3D12_LINEAR_ALGEBRA_DATATYPE {
+  D3D12_LINEAR_ALGEBRA_DATATYPE_SINT16          =  2, // ComponentType::I16
+  D3D12_LINEAR_ALGEBRA_DATATYPE_UINT16          =  3, // ComponentType::U16
+  D3D12_LINEAR_ALGEBRA_DATATYPE_SINT32          =  4, // ComponentType::I32
+  D3D12_LINEAR_ALGEBRA_DATATYPE_UINT32          =  5, // ComponentType::U32
+  D3D12_LINEAR_ALGEBRA_DATATYPE_FLOAT16         =  7, // ComponentType::F16
+  D3D12_LINEAR_ALGEBRA_DATATYPE_FLOAT32         =  8, // ComponentType::F32
+  D3D12_LINEAR_ALGEBRA_DATATYPE_INT8_T4_PACKED  = 16, // ComponentType::PackedS8x32
+  D3D12_LINEAR_ALGEBRA_DATATYPE_UINT8_T4_PACKED = 17, // ComponentType::PackedU8x32
+  D3D12_LINEAR_ALGEBRA_DATATYPE_UINT8           = 18, // ComponentType::U8
+  D3D12_LINEAR_ALGEBRA_DATATYPE_SINT8           = 19, // ComponentType::I8
+  D3D12_LINEAR_ALGEBRA_DATATYPE_E4M3            = 20, // ComponentType::F8_E4M3 (1 sign, 4 exp, 3 mantissa bits)
+  D3D12_LINEAR_ALGEBRA_DATATYPE_E5M2            = 21, // ComponentType::F8_E5M2 (1 sign, 5 exp, 2 mantissa bits)
 };
 
-typedef enum D3D12_COOPERATIVE_VECTOR_TIER
+typedef enum D3D12_LINEAR_ALGEBRA_MATRIX_VECTOR_TIER
 {
-    D3D12_COOPERATIVE_VECTOR_TIER_NOT_SUPPORTED,    
-    D3D12_COOPERATIVE_VECTOR_TIER_1_0,
-    D3D12_COOPERATIVE_VECTOR_TIER_1_1
+    D3D12_LINEAR_ALGEBRA_MATRIX_VECTOR_TIER_NOT_SUPPORTED,    
+    D3D12_LINEAR_ALGEBRA_MATRIX_VECTOR_TIER_1_0,
+    D3D12_LINEAR_ALGEBRA_MATRIX_VECTOR_TIER_1_1
 }
 
 // This struct may be augmented with more capability bits
 // as the feature develops
 typedef struct D3D12_FEATURE_DATA_D3D12_OPTIONSNN // NN tbd when implemented
 {
-    Out D3D12_COOPERATIVE_VECTOR_TIER CooperativeVectorTier;
+    Out D3D12_LINEAR_ALGEBRA_MATRIX_VECTOR_TIER LinearAlgebraMatrixVectorTier;
 } D3D12_FEATURE_DATA_D3D12_OPTIONSNN;
 
 // Used for MatrixVectorMulAdd intrinsic
-typedef struct D3D12_COOPERATIVE_VECTOR_PROPERTIES_INFERENCE
+typedef struct D3D12_LINEAR_ALGEBRA_MATRIX_VECTOR_PROPERTIES_INFERENCE
```

Comment:
@anupamachandra, is it accurate to put _VECTOR_ in this name?  As opposeed to just LINEAR_ALGEBRA_MATRIX_VECTOR_PROPERTIES_INFERENCE?

Same for TRAINING below.

Reply:
Yeah, we need MATRIX_VECTOR since the properties describe both the Input vector interpretation and the matrix.

Reply:
Same for training.